src/blocks/data.js

```diff
@@ -680,6 +699,67 @@ class Scratch3DataBlocks {
             }
         );
     }
+
+    renameCallback ({blockInfo, newName, varType}) {
+        const editingTarget = this.runtime.getEditingTarget();
+        const stage = this.runtime.getTargetForStage();
+        if (!editingTarget || !stage) return;
+
+        if (varType !== Variable.SCALAR_TYPE && varType !== Variable.LIST_TYPE) {
+            log.error(`Error processing invalid variable type: ${varType}`);
+            return;
+        }
+
+        const oldName = blockInfo.text;
+        const variable = editingTarget.lookupVariableByNameAndType(oldName, varType);
+
+        const fieldName = varType === Variable.SCALAR_TYPE ? 'VARIABLE' : 'LIST';
+
+        // Accumulate blocks that pertain to this extension
+        let allRelevantBlocks;
+        const visibleBlocks = editingTarget.blocks.getAllReferencesForVariable(variable);
```

Comment:
This is a call to a new function on the blocks container, which will return a list of all the variable/list blocks in that block container that reference a variable of the same name and type. There are other functions that are doing similar things for the regular core variables category, and I expect those to be removed post-extensionification of the variables category.